packages/block-editor/src/components/colors-gradients/panel-color-gradient-settings.js

```diff
@@ -136,23 +145,54 @@ export const PanelColorGradientSettingsInner = ( {
 			title={ showTitle ? titleElement : undefined }
 			{ ...props }
 		>
-			{ settings.map( ( setting, index ) => (
-				<ColorGradientControl
-					showTitle={ showTitle }
-					key={ index }
-					{ ...{
-						colors,
-						gradients,
-						disableCustomColors,
-						disableCustomGradients,
-						__experimentalHasMultipleOrigins,
-						__experimentalIsRenderedInSidebar,
-						enableAlpha,
-						...setting,
-					} }
-				/>
-			) ) }
-			{ children }
+			<ItemGroup isBordered isSeparated>
+				{ settings.map( ( setting, index ) => (
+					<Dropdown
```

Comment:
It looks like the addition of the Dropdown wrapper here adds an extra element around each item inside the itemgroup. So without the dropdown you have this markup:

<img width="491" alt="Screenshot 2021-12-03 at 11 10 35" src="https://user-images.githubusercontent.com/1204802/144585155-bfc4f244-9937-4252-88f4-594c0f8156f4.png">

With the extra wrapper, we have this:
<img width="472" alt="Screenshot 2021-12-03 at 11 10 55" src="https://user-images.githubusercontent.com/1204802/144585224-2e41d8ee-b1df-4381-8b6a-e5a2085b6d3f.png">

In terms of tabbing that doesn't seem to be a problem due to the tabindex properties. But the extra wrapper does mean this rule now is causing trouble:

<img width="318" alt="Screenshot 2021-12-03 at 11 12 09" src="https://user-images.githubusercontent.com/1204802/144585401-114f1d23-71e6-4192-a9b1-b047550dd016.png">

The rule is around the use of `border` for focus inside the itemgroup, making it transparent if it's the last child in an itemgroup. But because of the extra container, it's now adding transparent borders to every dropdown container which makes them all 1px taller. Compare this context:
<img width="399" alt="Screenshot 2021-12-03 at 11 09 32" src="https://user-images.githubusercontent.com/1204802/144585577-4d5e2fc0-79e6-4b8d-93da-1d8039855ce1.png">

... with that of global styles:
<img width="353" alt="Screenshot 2021-12-03 at 11 09 26" src="https://user-images.githubusercontent.com/1204802/144585589-2c56f81b-b0a7-44d5-84e0-bc523943d950.png">

My instinct tells me the solution is to refactor how focus works inside the itemgroup, to not rely on those advanced selectors, but perhaps instead just use a pseudo element. That would also let us avoid the use of transparent borders, which I think are problematic anyway due to them being made opaque in high contrast modes. 

But before I explore a fix for this in a separate PR, I wonder: @ciampo do you think my instinct above makes sense, or would there be a different or better way to fix this?

Reply:
@jorgefilipecosta it looks like the `ItemGroup` component is designed to have a very specific structure of `ItemGroup > Item` (like `ul > li`) and that the border styles are designed for the DOM structure to be pretty specific. As noted above the addition of the `Dropdown` is causing a few small issues since the structure becomes `ItemGroup > Dropdown > Item`, making the border rules target the wrong element.

It's not problematic enough to block this PR, I think, but it's also not ideal.

Just to see if it was possible, I tried to see if it would work to move the `Dropdown` inside the `Item`. Like so:

```
			<ItemGroup isBordered isSeparated>
				{ settings.map( ( setting, index ) => (
					<Item
						key={ index }
						className="block-editor-panel-color-gradient-settings__item"
					>
						<Dropdown
							contentClassName="block-editor-panel-color-gradient-settings__dropdown"
							renderToggle={ ( { onToggle } ) => {
								return (
									<HStack
										justify="flex-start"
										onClick={ onToggle }
									>
										<FlexItem>
											<ColorIndicator
												colorValue={
													setting.gradientValue ??
													setting.colorValue
												}
											/>
										</FlexItem>
										<FlexItem>{ setting.label }</FlexItem>
									</HStack>
								);
							} }
							renderContent={ () => (
								<ColorGradientControl
									showTitle={ false }
									{ ...{
										colors,
										gradients,
										disableCustomColors,
										disableCustomGradients,
										__experimentalHasMultipleOrigins,
										__experimentalIsRenderedInSidebar,
										enableAlpha,
										...setting,
									} }
								/>
							) }
						/>
					</Item>
				) ) }
			</ItemGroup>
```

It doesn't totally work, it replaces the `UnstyledButton` with the `Dropdown`. But is there a way we could make it work? Any thoughts?

Reply:
@jasmussen analysis is correct.

I've tried playing around with the markup a little, but I couldn't find any good solution — mainly trying to change the outline to something like:

```
ItemGroup
  Item
    Dropdown
      Button
        [content]
```

But that would mean that we'd need to apply most of the styles of `<Item>` to `<Button>` to achieve the same look&feel (and even then there could be some glitches)

I think that the best solution for now is to keep things as they are, and look into improving how `ItemGroup` and `Item` are styled directly at the source level.